packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/ReactBaseTextShadowNode.java

```diff
@@ -412,6 +413,12 @@ public void setNumberOfLines(int numberOfLines) {
     markUpdated();
   }
 
+  @ReactProp(name = ViewProps.MAXIMUM_NUMBER_OF_LINES, defaultInt = UNSET)
+  public void setMaxNumberOfLines(int numberOfLines) {
+    mMaxNumberOfLines = numberOfLines == 0 ? UNSET : numberOfLines;
```

Comment:
Why treat zero as unset?

Reply:
I think it was already the case before with numberOfLines. But seems to have sense. iOS NSTextContainer also uses it as the default value.